### DISCUSSION TITLE
Supabase flutter Wont Persist Auth State After Signin/Signup with Phone Number

### DISCUSSION THREAD

**Discussion by dukesx** [supabase/supabase#3970] [Questions]
Hi Everyone,

I am trying to sign my users up using Message bird Auth, Phone number + Password. Now, What i expect is that if the OTP is verified, I Should be Signed in and Session Stored to Hive (So that when i close the app and re open, i can visit the protected route) but it seems that i am unable to visit the protected route after closing and reopening the app. Any Help Here ?

Edit: 
I am in Debug Mode , currently developing the app. Debug is set to true in initialize.

**Comment by dshukertjr** [comment]
Hi @dukesx!

There is currently an open issue [here](https://github.com/supabase/gotrue-dart/issues/29) for this. It sounds like the issue might be fixed on the server end fairly soon. How soon do you plan to launch your app to the public?

**Reply by dukesx** [reply]
Hi @dshukertjr , Thank you for the response ! I am planning to launch it in 2 to 3 weeks (at most). I was really hoping if, i could just sign in and sign up by just using my number (WhatsApp like) and still persist session data for as long as JWT doesn't expire ? Is that possible ?

### Problem No 2
**Note:** I didn't understand what this line meant as well:

```
The user does not have a password therefore will need to sign in via this method each time they want to access your service.
```

Its on Message Bird tutorial page , Here -> https://supabase.com/docs/guides/auth/auth-messagebird#using-otp-as-a-passwordless-sign-in-mechanism

Does it mean that if i don't use password with phone auth, i won't get the session persistence  ?
